### ISSUE TITLE
Ignore string comparer requirement for Expression trees

### ISSUE THREAD

**Issue by molinch** [meziantou/Meziantou.Analyzer#301]
It's the same issue with CA1307

When building Expression trees that would then be consumed by EF.Core to generate SQL, you cannot specify the comparer for string operations.

So right now we disable CA1307 and MA0074 around these

**Comment by meziantou** [comment]
MA0074 should already exclude most cases with `Expression` because of EF Core ([source](https://github.com/meziantou/Meziantou.Analyzer/blob/main/src/Meziantou.Analyzer/Rules/UseStringComparisonAnalyzer.cs#L52-L55)).

Could you share a repro, so I can better understand your use case?

**Comment by molinch** [comment]
It's not really a blocker, today we do it so:
```c#
public IQueryable<Something> WithSomething(IQueryable<Something> query, IReadOnlyCollection<string> parts)
        {
#pragma warning disable CA1307 // Specify StringComparison for clarity: expression tree gets converted to SQL, the EF provider does not support specifying comparer
#pragma warning disable MA0006 // Similar reason
#pragma warning disable MA0074 // Similar reason

            return query
                .Where(parts
                        .Select(part => (Expression<Func<Something, bool>>)(s => s.SomeField.Contains(part)))
                        .CombineOr());

#pragma warning restore MA0006
#pragma warning restore MA0074
#pragma warning restore CA1307
        }
```

`CombineOr` is a helper we wrote that will transform all expressions into a chain of ORs, it's not relevant for the analyzer

**Comment by meziantou** [comment]
The analyzer only detects if the current invocation is in an argument of type `Expression`. I've added conversions to `Expression` in #302. It should fix your use-case.
